api/src/main/java/com/github/eyefloaters/console/api/model/PartitionInfo.java

```diff
@@ -35,39 +36,46 @@ public PartitionInfo(int partition, Node leader, List<Node> replicas, List<Node>
     }
 
     public static PartitionInfo fromKafkaModel(org.apache.kafka.common.TopicPartitionInfo info) {
-        Node leader = Node.fromKafkaModel(info.leader());
         List<Node> replicas = info.replicas().stream().map(Node::fromKafkaModel).toList();
-        List<Node> isr = info.isr().stream().map(Node::fromKafkaModel).toList();
-        return new PartitionInfo(info.partition(), leader, replicas, isr);
+        Integer leader = info.leader().id();
+        List<Integer> isr = info.isr().stream().map(org.apache.kafka.common.Node::id).toList();
+        return new PartitionInfo(info.partition(), replicas, leader, isr);
     }
 
-    static Either<OffsetInfo, Error> offsetOrError(Either<OffsetInfo, Throwable> offset) {
+    static <P> Either<P, Error> primaryOrError(Either<P, Throwable> offset, String message) {
         return offset.ifPrimaryOrElse(
                 Either::of,
-                thrown -> Error.forThrowable(thrown, "Unable to fetch partition offset"));
+                thrown -> Error.forThrowable(thrown, message));
     }
 
     public void addOffset(String key, Either<OffsetInfo, Throwable> offset) {
         if (this.offsets == null) {
             this.offsets = new LinkedHashMap<>(4);
         }
 
-        this.offsets.put(key, offsetOrError(offset));
+        this.offsets.put(key, primaryOrError(offset, "Unable to fetch partition offset"));
+    }
+
+    public void addReplicaInfo(int nodeId, Either<ReplicaInfo, Throwable> log) {
```

Comment:
It seems pretty surprising that this method mutates the node. Are these public mutator methods unavoidable?

Reply:
I think the mutators will be unavoidable mainly because the main structural parts of the model are constructed earlier (e.g. using data from `describeTopics`), and the missing pieces are filled-in afterwards. This may be improved if we change how replicas are modeled.

Reply:
The `addReplicaInfo` method and params were renamed to:
```java
setReplicaLocalStorage(int nodeId, Either<ReplicaLocalStorage, Throwable> storage)
```